golem_sci/implementation.py

```diff
@@ -588,19 +588,33 @@ def _pull_eth_subscription_events(self) -> None:
             subs = self._eth_subscriptions.copy()
         for sub in subs:
             try:
-                if sub.last_pulled_block >= self._confirmed_block:
-                    continue
-                transfers = self._find_incoming_eth_transfers(
-                    sub.last_pulled_block,
-                    self._confirmed_block,
-                    sub.address,
-                )
-                for t in transfers:
-                    self._on_event(t, sub.cb)
-                sub.last_pulled_block = self._confirmed_block
+                try:
+                    if sub.last_pulled_block >= self._confirmed_block:
+                        continue
+                    transfers = self._find_incoming_eth_transfers(
+                        sub.last_pulled_block,
+                        self._confirmed_block,
+                        sub.address,
+                    )
+                    for t in transfers:
+                        self._on_event(t, sub.cb)
+                    sub.last_pulled_block = self._confirmed_block
+                except ValueError as e:
```

Comment:
You may reuse `_is_jsonrpc_error` function from this file to check whether it's the error we're interested in. And just log the message regardless of what's inside.

Reply:
\o/